### PR TITLE
Gstreamer transcode 1to N tests added

### PR DESCRIPTION
**Pull request by wangzj0601**
1. Support transcode 1toN tests
  2. It supports AVC/HEVC to AVC/HEVC/MPEG2/MJPEG transcoding for N times

Signed-off-by: Wang Zhanjun <zhanjunx.wang@intel.com>

### CODE REVIEW

**Comment by wangzj0601** [comment]
can you help have a review, thanks @haribommi  @uartie

File: test/gst-msdk/transcode/transcoder.py [hunk 1]
```diff
@@ -69,8 +70,6 @@ def validate_spec(self):
     from slash.utils.pattern_matching import Matcher
 
     assert len(self.outputs), "Invalid test case specification, outputs data empty"
-    if len(self.outputs) > 1 or self.outputs[0].get("channels", 1) > 1:
-      slash.skip_test("transcode 1ton tests not supported")
```
Comment (uartie) [inline test/gst-msdk/transcode/transcoder.py]: We also need to validate all of the outputs (requirements, platforms, mode, etc.) below.  See https://github.com/intel/vaapi-fits/blob/1d853934d6d25fd4164b3339dddae729438feeb2/test/ffmpeg-vaapi/transcode/transcoder.py#L76 as example.

File: test/gst-msdk/transcode/transcoder.py [hunk 2]
```diff
@@ -161,23 +168,27 @@ def transcode(self):
...
-      " dump-location={}".format(self.ofile, refyuv_decode_arg, yuv))
-    self.check_metrics(yuv)
-
-  def check_metrics(self, yuv):
+    for n, output in enumerate(self.outputs):
+      for channel in xrange(output.get("channels", 1)):
+        encoded = self.goutputs[n][channel]
+        yuv = get_media()._test_artifact(
+          "{}_{}_{}.yuv".format(self.case, n, channel))
+        refyuv_decode_arg = self.gen_refyuv_decode_commands(self.outputs[0]["codec"], "hw")
```
Comment (uartie) [inline test/gst-msdk/transcode/transcoder.py]: `self.outputs[0]` is wrong.  Each output might use a different codec but you are using the codec from the first output for all outputs.  For example, `self.output[0]` might be `avc` and `self.output[1]` might be `mjpeg`.  You need to use the codec in current loop iteration of `output`.

File: test/gst-msdk/transcode/transcoder.py (hunk 1 above)
Comment (wangzj0601) [reply test/gst-msdk/transcode/transcoder.py]: sorry, have fixed it.

File: test/gst-msdk/transcode/transcoder.py (hunk 2 above)
Comment (wangzj0601) [reply test/gst-msdk/transcode/transcoder.py]: sorry, have fixed it.

**Comment by uartie** [comment]
Please update vaapi-fits example test baselines for gst transcode.

**Comment by wangzj0601** [comment]
> Please update vaapi-fits example test baselines for gst transcode.

have done it, thanks

**Comment by uartie** [comment]
Whoops.  Looks like you didn't provide i965 baseline.  Please submit in new pull-request.

**Pull request merged**